File: Annadata/app/src/main/java/com/mananhirak/annadata/fragments/BuyMarketFragment.java
package com.mananhirak.annadata.fragments;

import android.content.Context;
import android.os.Bundle;

import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;

import com.firebase.ui.firestore.FirestoreRecyclerOptions;
import com.google.android.gms.tasks.OnSuccessListener;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;
import com.google.firebase.firestore.DocumentSnapshot;
import com.google.firebase.firestore.FirebaseFirestore;
import com.google.firebase.firestore.Query;
import com.mananhirak.annadata.R;
import com.mananhirak.annadata.dbhelpers.MARKET;
import com.mananhirak.annadata.foodcycle.MarketAdapter;
import com.mananhirak.annadata.idtag.FireTag;
import com.mananhirak.annadata.idtag.Mtag;


public class BuyMarketFragment extends Fragment {

    private Context context;
    Query query;
    RecyclerView recyclerView;
    FirebaseFirestore FIRE_db;
    MarketAdapter marketAdapter;
    FirebaseUser user;
    View v;
    private String SellerBannerId;

    public BuyMarketFragment() {
        // Required empty public constructor
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null)
            SellerBannerId = getArguments().getString("TiffinLinkSellerId");
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        v =inflater.inflate(R.layout.fragment_buy_market, container, false);

        context =getContext();
        recyclerView=v.findViewById(R.id.rcycle_fb_marketcycle);
        recyclerView.setLayoutManager(new LinearLayoutManager(context));

        Log.d(Mtag.Tag, "onCreateView BuyFragment: 93 ");

        user = FirebaseAuth.getInstance().getCurrentUser();
        FIRE_db=FirebaseFirestore.getInstance();

        FIRE_db.collection(FireTag.FireUser)
                .document(user.getUid())
                .get()
                .addOnSuccessListener(new OnSuccessListener<DocumentSnapshot>() {
                    @Override
                    public void onSuccess(DocumentSnapshot documentSnapshot) {
                        SetQuery((String)documentSnapshot.get("address"));
                    }
                });

        return v;
    }

    private  void SetQuery(String address){

        Log.d(Mtag.Tag, "SetQuery: 113 "+address);


        if(SellerBannerId!=null){
            query = FirebaseFirestore.getInstance()
                    .collection(FireTag.FireMarket)
                    .whereEqualTo("retailer_ID",SellerBannerId);
        }else{
            query = FirebaseFirestore.getInstance()
                    .collection(FireTag.FireMarket)
                    .whereEqualTo("city_NAME",address)
                    .whereNotEqualTo("retailer_ID",user.getUid())
                    .whereEqualTo("no_FOOD_LEFT",false);

        }
        SetAdapter();

    }

    private void SetAdapter(){
        FirestoreRecyclerOptions<MARKET> options= new FirestoreRecyclerOptions.Builder<MARKET>()
                .setQuery(query,MARKET.class)
                .build();
        Log.d(Mtag.Tag, "onCreateView BuyFragment: 128 ");

        marketAdapter=new MarketAdapter(options,context);
        recyclerView.setAdapter(marketAdapter);
        marketAdapter.startListening();
    }

    @Override
    public void onStop() {
        super.onStop();
        if(marketAdapter!=null){
            marketAdapter.stopListening();
        }
    }
}

File: Annadata/app/src/main/java/com/manan/annadata/fragments/BuyMarketFragment.kt
package com.manan.annadata.fragments

import android.content.Context
import com.google.firebase.auth.FirebaseUser
import com.google.firebase.firestore.FirebaseFirestore
import androidx.recyclerview.widget.RecyclerView
import com.manan.annadata.foodcycle.MarketAdapter
import android.os.Bundle
import android.util.Log
import com.manan.annadata.fragments.PersonalFragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.manan.annadata.R
import androidx.recyclerview.widget.LinearLayoutManager
import com.google.firebase.auth.FirebaseAuth
import com.manan.annadata.idtag.FireTag
import com.google.android.gms.tasks.OnSuccessListener
import com.google.firebase.firestore.DocumentSnapshot
import com.firebase.ui.firestore.FirestoreRecyclerOptions
import com.google.firebase.firestore.Query
import com.manan.annadata.dbhelpers.MARKET
import com.manan.annadata.idtag.Mtag

class BuyMarketFragment : Fragment() {
    var query: Query? = null
    var recyclerView: RecyclerView? = null
    var FIRE_db: FirebaseFirestore? = null
    var marketAdapter: MarketAdapter? = null
    var user: FirebaseUser? = null
    var v: View? = null
    private var SellerBannerId: String? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (arguments != null) SellerBannerId = requireArguments().getString("TiffinLinkSellerId")
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        v = inflater.inflate(R.layout.fragment_buy_market, container, false)

        recyclerView = v!!.findViewById(R.id.rcycle_fb_marketcycle)
        recyclerView!!.setLayoutManager(LinearLayoutManager(context))
        Log.d(Mtag.Tag, "onCreateView BuyFragment: 93 ")
        user = FirebaseAuth.getInstance().currentUser
        FIRE_db = FirebaseFirestore.getInstance()
        FIRE_db!!.collection(FireTag.FireUser)
            .document(user!!.uid)
            .get()
            .addOnSuccessListener { documentSnapshot ->
                SetQuery(
                    documentSnapshot["address"] as String?
                )
            }
        return v
    }

    private fun SetQuery(address: String?) {
        Log.d(Mtag.Tag, "SetQuery: 113 $address")
        query = if (SellerBannerId != null) {
            FirebaseFirestore.getInstance()
                .collection(FireTag.FireMarket)
                .whereEqualTo("retailer_ID", SellerBannerId)
        } else {
            FirebaseFirestore.getInstance()
                .collection(FireTag.FireMarket)
                .whereEqualTo("city_NAME", address)
                .whereNotEqualTo("retailer_ID", user!!.uid)
                .whereEqualTo("no_FOOD_LEFT", false)
        }
        SetAdapter()
    }

    private fun SetAdapter() {
        val options = FirestoreRecyclerOptions.Builder<MARKET>()
            .setQuery(query!!, MARKET::class.java)
            .build()
        Log.d(Mtag.Tag, "onCreateView BuyFragment: 128 ")
        marketAdapter = context?.let { MarketAdapter(options, it) }
        recyclerView!!.adapter = marketAdapter
        marketAdapter!!.startListening()
    }

    override fun onStop() {
        super.onStop()
        if (marketAdapter != null) {
            marketAdapter!!.stopListening()
        }
    }
}